modules/cnf-configuring-llc-aware-cpu-pinning.adoc

```diff
@@ -0,0 +1,264 @@
++
+.Example `kc-llc-aware.yaml` file
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: llc-aware-kubeletconfig
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      machineconfiguration.openshift.io/role: worker-cnf <1> 
+  kubeletConfig:
+    cpuManagerPolicy: "static" <2>
+    cpuManagerPolicyOptions:
+        # full-pcpus-only: "true" Is this needed?
+        prefer-align-cpus-by-uncorecache: "true" <3>
+#...
+----
+<1> Selects the nodes in the `worker-cnf` machine config pool.
+<2> Set the `cpuManagerPolicy` to `static`.
+<3> Enable the `prefer-align-cpus-by-uncorecache` option for LLC-aware scheduling by the CPU Manager.
++
+[NOTE]
+====
+The node will restart after you applying the `KubeletConfig` resource.
+====
+
+.. Apply the `KubeletConfig` resource by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f kc-llc-aware.yaml
+----
++
+.Example output
+[source,terminal]
+----
+machineconfig.machineconfiguration.openshift.io/99-worker-openshift-llc-alignment created
+----
+
+.Verification
+
+. Create a `Pod` resource to test the configuration:
+
+.. Create a YAML file that defines a `Pod` resource:
++
+.Example `pod-llc-aware.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: llc-aware-pod
+spec:
+  containers:
+  - name: centos-container
+    image: quay.io/centos/centos:stream9 <1>
+    command: ["/bin/bash", "-c", "sleep infinity"]
+    resources:
+      limits:
+        cpu: "1" <2>
+        memory: "100Mi"
+      requests:
+        cpu: "1" <3>
+        memory: "100Mi"
+  nodeSelector:
+    node-role.kubernetes.io/worker-cnf: "" <4>
+----
+<1> This example uses a CentOS image to test the configuration.
+<2> The `cpu` field specifies the number of CPU cores to use.
+<3> The `requests` field specifies the number of CPU cores to request. 
+<4> The `nodeSelector` field selects the node with the `worker-cnf` label.
+
+.. Apply the `Pod` resource by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f pod-llc-aware.yaml
+----
++
+.Example output
+[source,terminal]
+----
+pod/llc-aware-pod created
+----
+
+. Remotely enter the container at the command line to check the CPU assignment by running the following command:
++
+[source,terminal]
+----
+$ oc exec -it llc-aware-pod --container centos-container -- /bin/bash
+----
++
+[source,terminal]
+----
+bash-5.1# cat /sys/fs/cgroup/cpuset.cpus
+----
++
+.Example output
+[source,terminal]
+----
+2
+----
+* The output shows the CPU cores assigned to the container. Verify the cores are aligned with the LLC cache according to the split LLC architecture of the CPU.
+
+. Check that the enablement file was created on the target node by running the following commands:
++
+[source,terminal]
+----
+$ oc debug node/<node_name>
+----
++
+[source,terminal]
+----
+sh-5.1# chroot /host
+sh-5.1# ls /etc/kubernetes/ | grep openshift-llc-alignment
```

Comment:
or even
```
stat /etc/kubernetes/openshift-llc-alignment
```
any output means the file is present
error means the file is absent
another option
```
ls /etc/kubernetes/openshift-llc-alignment
```